app/process_proposal.go

```diff
@@ -115,6 +118,9 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 			return reject()
 		}
 
+		// Set the tx size on the context before calling the AnteHandler
+		sdkCtx = sdkCtx.WithTxBytes(tx)
```

Comment:
[same question] does this need to be gated behind a version check?

Can this be hoisted up so that it only needs to happen once in this file? Perhaps after this line:

```go
		sdkTx, err := app.txConfig.TxDecoder()(tx)
		// Set the tx bytes in the context for app version v3 and greater
		if sdkCtx.BlockHeader().Version.App >= 3 {
			sdkCtx = sdkCtx.WithTxBytes(tx)
		}
		// rest of file
```

Reply:
referencing this comment https://github.com/celestiaorg/celestia-app/pull/3909#discussion_r1796032074

we need a gate for process proposal, we don't need a gate prepare proposal

Reply:
https://github.com/celestiaorg/celestia-app/pull/3909/commits/a873099ce3d6f205a1bcf5c6768c3a9926852a7f